secure_message/resources/messages.py

```diff
@@ -76,20 +77,34 @@ def post(self):
             if last_modified is False:
                 return Response(response="Draft has been modified since last check", status=409, mimetype="text/html")
 
-        message = MessageSchema().load(post_data)
+        post_data['from_internal'] = g.user.is_internal
+        message = self._validate_post_data(post_data)
+
         if message.errors == {}:
-            return self._message_save(message, is_draft, draft_id)
+            self._message_save(message, is_draft, draft_id)
+            if is_draft:
+                Modifier().del_draft(draft_id)
+            # listener errors are logged but still a 201 reported
+            MessageSend._alert_listeners(message.data)
+            resp = jsonify({'status': '201', 'msg_id': message.data.msg_id, 'thread_id': message.data.thread_id})
+            resp.status_code = 201
```

Comment:
Status is being set twice.

Reply:
No, the jsonify sets status_code to 200 , we want a 201 returned to the user to indicate that its been added

Reply:
Jsonify just wraps the JSON library with some helper methods that output a flask.Response object. I can see in line 89 that `"status": "201"` and 
 on line 90 `"status_code": "201"`. If you need to set both fields to the same value then there is something wrong, either here or elsewhere in the service.